### PR TITLE
feat: preserve attachments in blob reports

### PR DESCRIPTION
**Pull request by yury-s**
Attachments with `path` are stored as `resources/sha1.ext` zip entries and extracted under `report-dir/tmp` when merging. This way normal fs.readFile keeps working as before even thought the file path is different. The clients should rely on `attachment.name` instead of `attachment.path` when deriving user visible titles in the UI. If this turns out not to be the case we can reconsider later.

#10437

### CODE REVIEW

File: packages/playwright-test/src/reporters/blob.ts [hunk 1]
```diff
@@ -83,66 +112,103 @@ export class BlobReporter extends TeleReporterEmitter {
...
-    list: ListReporter,
-    github: GitHubReporter,
-    json: JSONReporter,
-    junit: JUnitReporter,
-    null: EmptyReporter,
-    html: HtmlReporter,
-    blob: BlobReporter,
-  };
-  reporterName ??= 'list';
+  const resourceDir = path.join(dir, 'temp');
```
Comment (dgozman) [inline packages/playwright-test/src/reporters/blob.ts:115]: Let's do this inside `os.tmpdir()`

File: packages/playwright-test/src/reporters/blob.ts [hunk 2]
```diff
@@ -83,66 +112,103 @@ export class BlobReporter extends TeleReporterEmitter {
...
+        const fileName = path.join(resourceDir, entryName);
+        await fs.promises.mkdir(path.dirname(fileName), { recursive: true });
+        await fs.promises.writeFile(fileName, content);
+      }
+    }
+  }
+  return reports;
+}
+
+function patchAttachmentPaths(events: JsonEvent[], rootDir: string) {
```
Comment (dgozman) [inline packages/playwright-test/src/reporters/blob.ts:178]:
```suggestion
function patchAttachmentPaths(events: JsonEvent[], resourceDir: string) {
```

File: packages/playwright-test/src/reporters/blob.ts [hunk 3]
```diff
@@ -83,66 +112,103 @@ export class BlobReporter extends TeleReporterEmitter {
...
-    reporter = new defaultReporters[reporterName as keyof typeof defaultReporters](options);
-  } else {
-    const reporterConstructor = await loadReporter(config, reporterName);
-    reporter = new reporterConstructor(options);
-  }
+    const arg = config.config.reporter.find(([reporter, arg]) => reporter === reporterName)?.[1];
+    const options = {
+      ...arg,
+      configDir: process.cwd(),
+      outputFolder: path.join(dir, 'merged-report')
```
Comment (dgozman) [inline packages/playwright-test/src/reporters/blob.ts:139]: This should come from the config instead, and it should just work (tm).

File: packages/playwright-test/src/reporters/blob.ts [hunk 4]
```diff
@@ -83,66 +112,103 @@ export class BlobReporter extends TeleReporterEmitter {
...
+    const events = mergeEvents(shardReports);
+    patchAttachmentPaths(events, resourceDir);
 
-  const arg = config.config.reporter.find(([reporter, arg]) => reporter === reporterName)?.[1];
-  const options = {
-    ...arg,
-    configDir: process.cwd(),
-    outputFolder: dir
-  };
+    const defaultReporters: {[key in BuiltInReporter]: new(arg: any) => Reporter} = {
```
Comment (dgozman) [inline packages/playwright-test/src/reporters/blob.ts:122]: Let's try to reuse `createReporter` function.

**Review by dgozman — APPROVED** [review]

File: packages/playwright-test/src/reporters/blob.ts (hunk 4 above)
Comment (yury-s) [reply packages/playwright-test/src/reporters/blob.ts:122]: I'll try to do that in a follow-up as agreed offline.

**Pull request merged**